plugins/quota-aware-fs/src/main/java/org/elasticsearch/fs/quotaaware/QuotaAwarePath.java

```diff
@@ -179,7 +180,11 @@ public Path toRealPath(LinkOption... options) throws IOException {
     @SuppressForbidden(reason = "Implementation required by super type")
     @Override
     public File toFile() {
```

Comment:
In `master` we simply don't override this method at all, instead implicitly delegating to the default implementation on the `Path` interface. Is there a reason we can't just do that here instead of duplicating this?

Reply:
As I understand it, `7.x` compiles against an older version of the JDK, where `Path.toFile()` doesn't have a default implementation, so we **have** to provide an implementation (Cloud has the same issue), even though the runtime JDK provides one. All I did was take the default implementation and paste it into the `7.x` source. That means that the same code ends up being run in any case, except now a `delegate.toFile()` call is removed, and that seems to change the context enough to fix the problem.

I managed to replicate the problem that @mieciu found by running the an ES configuration from him against `7.x`. I then iterated on the plugin source until the problem went away.

Reply:
Ah yes, you are absolutely right. So yes, in `7.x` the minimum runtime version is Java 8 (vs Java 11 for master), so when we compile (although we compile using a  Java 14 JDK) we use `--release 8` so we "effectively" compile against the Java 8 APIs. The _bundled_ runtime JDK includes an implementation here, but we support folks running with a Java 8 JRE so we can't rely on that.

LGTM.